hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/DBConfigFromFile.java

```diff
@@ -103,37 +108,87 @@ public static String getOptionsFileNameFromDB(String dbFileName) {
    * control OzoneManager.db configs from a file, we need to create a file
    * called OzoneManager.db.ini and place that file in $OZONE_DIR/etc/hadoop.
    *
-   * @param dbFileName - The DB File Name, for example, OzoneManager.db.
-   * @param cfDescs - ColumnFamily Handles.
+   * @param dbPath - The DB File Name, for example, OzoneManager.db.
    * @return DBOptions, Options to be used for opening/creating the DB.
    */
-  public static ManagedDBOptions readFromFile(String dbFileName,
-      List<ColumnFamilyDescriptor> cfDescs) throws RocksDatabaseException {
-    Preconditions.checkNotNull(dbFileName);
-    Preconditions.checkNotNull(cfDescs);
-    Preconditions.checkArgument(!cfDescs.isEmpty());
+  public static ManagedDBOptions readDBOptionsFromFile(Path dbPath) throws RocksDBException {
+    List<ColumnFamilyDescriptor> descriptors = new ArrayList<>();
+    try {
+      return readFromFile(dbPath, descriptors);
+    } finally {
+      //readDBOptions will be freed once the store using it is closed, but the descriptors need to be closed.
+      closeDescriptors(descriptors);
+    }
+  }
 
-    //TODO: Add Documentation on how to support RocksDB Mem Env.
-    Env env = Env.getDefault();
-    ManagedDBOptions options = null;
-    File configLocation = getConfigLocation();
-    if (configLocation != null &&
-        StringUtils.isNotBlank(configLocation.toString())) {
-      Path optionsFile = Paths.get(configLocation.toString(),
-          getOptionsFileNameFromDB(dbFileName));
-
-      if (optionsFile.toFile().exists()) {
-        options = new ManagedDBOptions();
-        try {
-          OptionsUtil.loadOptionsFromFile(optionsFile.toString(),
-              env, options, cfDescs, true);
-
-        } catch (RocksDBException rdEx) {
-          throw new RocksDatabaseException("Failed to loadOptionsFromFile " + optionsFile, rdEx);
-        }
+  public static ManagedColumnFamilyOptions readCFOptionsFromFile(Path dbPath, String cfName) throws RocksDBException {
+    List<ColumnFamilyDescriptor> descriptors = new ArrayList<>();
+    String defaultColumnFamilyString = StringUtils.toEncodedString(DEFAULT_COLUMN_FAMILY, StandardCharsets.UTF_8);
+    String validatedCfName = StringUtils.isEmpty(cfName) ? defaultColumnFamilyString : cfName;
+    ManagedColumnFamilyOptions resultCfOptions = null;
+    try (ManagedDBOptions ignored = readFromFile(dbPath, descriptors)) {
+      ColumnFamilyDescriptor descriptor = descriptors.stream()
+          .filter(desc -> StringUtils.toEncodedString(desc.getName(), StandardCharsets.UTF_8).equals(validatedCfName))
+          .findAny().orElse(null);
+      if (descriptor != null) {
+        resultCfOptions = new ManagedColumnFamilyOptions(descriptor.getOptions());
       }
+    } finally {
+      closeDescriptors(descriptors);
+    }
+    return resultCfOptions;
+  }
+
+  private static ManagedDBOptions readFromFile(Path dbPath, List<ColumnFamilyDescriptor> descriptors)
```

Comment:
This function can return a AutoCloseable wrapper which would close the descriptors when close is called. Moreover we are always creating a new ArrayList<>() for descriptors so it would be better if just initialize descriptor inside this function